brave/brave-ios
# Fix#2980: Educational in-product Notification

This PR is adding functionality to educational in-product notifications.

The added functionality can be states as follows.

Notification types will be shown 

- First time blocking
- 20+ trackers & ads blocked on this page.
- HTTPS Upgrade triggered
- A Streaming site is loaded and ad is blocked

And also Brave Benchmark Tiers are added so every tier user will be shown a notification regarding to which level they are and sharing options for Issue #2981 

Please refer to respected documents for details.

## Summary of Changes

This pull request fixes #2980

## Submitter Checklist:


## Test Plan:

- Load fresh installed Brave Browser
- Type www.google.com in address bar and check Type 1  - "Trackers & ads blocked on this page." pop-over is displayed.
- Press "Take a look" button and check Brave Shields pop-over is displayed.
- Load a URL from Vimeo (https://vimeo.com/search?q=test) and check Type 2 - "Ads are blocked while watching videos on this website" pop-over is displayed.
- Kill the application and start again. Type 2 - 3 - 4 (blue background) pop-overs can be shown only once per application session (repeat this every time you want to check a different type of notification pop-over.
-  Load a URL from a website that has lots of tracker like  (www.buzzfeed.com), check Type 3 - "10+ trackers & ads blocked on this page." pop-over is displayed.
- Repeat the previous step (Kill the application and start again)
-  Load a URL from a website that triggers from a website that triggers https upgrade like (www.cnn.com or https://imagery.pragprog.com), check Type 4 - "Your connection is now encrypted." pop-over is displayed.

## Screenshots:


![slack-imgs](https://user-images.githubusercontent.com/6643505/105050806-28901180-5a3c-11eb-8f30-3cb5d467214f.png)



## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
